packages/cdktf-cli/lib/checkpoint.ts

```diff
@@ -0,0 +1,80 @@
+import https = require('https');
+import { format } from 'url';
+import { v4 as uuidv4 } from 'uuid';
+import * as os from 'os';
+
+const BASE_URL = `https://checkpoint-api.hashicorp.com/v1/`;
+
+const VALID_ERROR_CODES = [200, 201];
+
+export interface ReportParams {
+    dateTime?: Date;
+    arch?: string;
+    os?: string;
+    payload: {};
+    product: string;
+    runID?: string;
+    version?: string;
+}
+
+async function post(url: string, data: string) {
+    return new Promise<any>((ok, ko) => {
+        const req = https.request(format(url), {
+            headers: {
+                'Accept': 'application/json',
+                'Content-Length': data.length,
+                'User-Agent': 'HashiCorp/cdktf-cli'
+            },
+            method: 'POST'
+        }, res => {
+            if (res.statusCode) {
+                const statusCode = res.statusCode;
+                if (!VALID_ERROR_CODES.includes(statusCode)) {
+                    return ko(new Error(res.statusMessage));
+                }
+            }
+            const data = new Array<Buffer>();
+            res.on('data', chunk => data.push(chunk));
+
+            res.once('error', err => ko(err));
+            res.once('end', () => {
+                return ok();
+            });
+        });
+
+        req.write(data);
+
+        req.end();
+    })
+}
+
+export async function ReportRequest(reportParams: ReportParams): Promise<void> {
+    // we won't report when checkpoint is disabled.
+    if (process.env.CDKTF_CHECKPOINT_DISABLE == "1") {
```

Comment:
should this be set for our tests - so pretty much for all CI jobs?

Reply:
Yes, I think that would help give with generating accurate telemetry.